src/ast.rs

```diff
@@ -290,6 +292,30 @@ impl Query {
     }
 }
 
+/// A query tied to a particular universe through symbols
+pub struct UniverseQuery<'a> {
```

Comment:
Can you move this type into the `textual` module, or perhaps `universe` otherwise? It feels a bit out of place here, since the rest of the `ast` module is purely about a more direct representation of the various syntactic structures, but not really concerned with managing symbols.

(So I'd see it more as part of the lower-level abstractions that shouldn't need to deal with this association between AST nodes and the underlying symbol storage).

Reply:
Done.